helix-term/src/ui/editor.rs

```diff
@@ -170,7 +170,20 @@ impl EditorView {
         theme: &Theme,
     ) {
         let editor_rulers = &editor.config().rulers;
-        let ruler_theme = theme.get("ui.virtual.ruler");
+        let default_style = Style {
+            bg: Some(Color::Red),
+            ..Default::default()
+        };
+        let ruler_theme = theme
+            .try_get("ui.virtual.ruler")
+            .map(|style| {
+                if let None = style.bg {
+                    default_style
+                } else {
+                    style
+                }
+            })
```

Comment:
```suggestion
            .filter(|style| style.bg.is_some())
```

Reply:
With this map/filter though, you can't create a ruler that is `fg` only, and `fg`-only rulers are kinda nice because they aren't visible until text crosses them.

I think having a louder default is a good idea. If we remove this filter line we'll still have good fallback behavior when a theme doesn't define `ui.virtual.ruler` or `ui.virtual` (or `ui`) at all.

Reply:
The reason I added this is because pretty much all themes define `ui.virtual` after #1802 and pretty much all of them define it as just foreground color, which makes it fairly unobvious until the text reaches it as you mention

Reply:
What about adding a `get_exact` to theme that doesn't fallback to `ui.virtual`?

Reply:
`get_exact` would make this scope special. I would prefer that all scopes have fallback behavior for consistency's sake.

Having rulers themed with `ui.ruler` might be preferrable since `ui.virtual.whitespace` and `ui.virtual.ruler` end up conflicting which one wants `fg` and which wants `bg`. This would be a breaking change though, although I suppose that could be mitigated by adding code here to fall back to `ui.virtual.ruler` and logging a warning.

Alternatively, I think it a PR that adds `ui.virtual.ruler` to all in-repo themes that don't already have it would go a long way towards alleviating this.

Reply:
> `get_exact` would make this scope special. I would prefer that all scopes have fallback behavior for consistency's sake.

Yeah I was worried about that, makes sense to avoid it

> Having rulers themed with `ui.ruler` might be preferrable since `ui.virtual.whitespace` and `ui.virtual.ruler` end up conflicting which one wants `fg` and which wants `bg`. This would be a breaking change though, although I suppose that could be mitigated by adding code here to fall back to `ui.virtual.ruler` and logging a warning.

That makes sense, but I am not really sure if there's precedence I can look at for how to emit that warning.

> Alternatively, I think it a PR that adds `ui.virtual.ruler` to all in-repo themes that don't already have it would go a long way towards alleviating this.

Yeah I agree, I was considering doing that but realized it might still trip people adding themes in the future or people using custom themes so I tried this and it worked well enough so I wanted to see what people thought of this idea

Reply:
Actually I think switching to

```rust
let ruler_theme = theme
    .try_get("ui.ruler")
    .unwrap_or_else(|| Style::default().bg(Color::Red));
```

might be the best option: it'll be very noticeable when a theme doesn't have `ui.ruler` and it won't fall back to `ui.virtual`.

Reply:
I just thought of another alternative that lets us have a loud default for everyone without breaking changes. How about we update all current themes from `"ui.virtual"` to `"ui.virtual.whitespace"` and then we can just filter by whether there is any style or not, in most cases there won't be so it will default to red

Reply:
Yeah that's a good idea 👍

Reply:
Should be good now!